ipam/http.go

```diff
@@ -64,6 +64,26 @@ func (alloc *Allocator) handleHTTPAllocate(dockerCli *docker.Client, w http.Resp
 	fmt.Fprintf(w, "%s/%d", addr, subnet.PrefixLen)
 }
 
+func (alloc *Allocator) handleHTTPClaim(w http.ResponseWriter, ident string,
+	cidr address.CIDR, isContainer, noErrorOnUnknown bool) {
+
+	closedChan := w.(http.CloseNotifier).CloseNotify()
+	hasBeenCanceled := func() bool {
+		select {
+		case <-closedChan:
+			return true
+		default:
+			return false
+		}
+	}
+	err := alloc.Claim(ident, cidr, isContainer, noErrorOnUnknown, hasBeenCanceled)
+	if err != nil {
+		badRequest(w, fmt.Errorf("Unable to claim: %s", err))
+		return
+	}
+	w.WriteHeader(204)
+}
+
```

Comment:
Is there a reason why the above is so different from `handleHTTPAllocate`? i.e.

- allocate does a checkAlive inside hasBeenCancelled but claim does not
- allocate suppresses errorCancelled errors but claim does not
- allocate uses an anonymous function for the cancellation check but claim does not